builder/FindGlobals.cmake

```diff
@@ -91,8 +92,8 @@ if( Linux OR Darwin )
   set(c_warnings "-Wall -Wformat -Wformat-security")
   set(cxx_warnings "${c_warnings} -Wnon-virtual-dtor")
 
-  set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -pipe -fPIC ${c_warnings} ${no_warnings}")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pipe -fPIC ${cxx_warnings} ${no_warnings}")
+  set(CMAKE_C_FLAGS   "${CMAKE_FLAGS} ${CMAKE_C_FLAGS} -pipe -fPIC ${c_warnings} ${no_warnings}")
```

Comment:
Can you, please, avoid using CMAKE_FLAGS name for the variable? I suspect that this name fakes cmake native variable while actually cmake never defines such and uses CMAKE_<LANG>_FLAGS type of things. Please, set language specific flags only, i.e. CMAKE_C_FLAGS and CMAKE_CXX_FLAGS. To avoid setting everything twice, use temporary variable like:
set(common_flags -DUNIX)
```
list(APPEND common_flags -D__USE_LARGEFILE64)
...
set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -pipe -fPIC ${c_warnings} ${no_warnings} ${common_flags} ")
set(CMAKE_CXX_FLAGS   "${CMAKE_CXX_FLAGS} -pipe -fPIC ${c_warnings} ${no_warnings} ${common_flags} ")
```

Also, please, avoid using self-defined append function, use cmake native set and list. Our self definition does not actually makes much sense - it does not simplify anything, just obscures what's going on.

Reply:
done